neqo-transport/src/crypto.rs

```diff
@@ -584,7 +584,7 @@ impl CryptoDxAppData {
         }
         let next_secret = Self::update_secret(self.cipher, &self.next_secret)?;
         Ok(Self {
-            dx: self.dx.next(&next_secret, self.cipher),
+            dx: self.dx.next(&self.next_secret, self.cipher),
```

Comment:
I keep asking myself whether Rust's semantics could have helped here at all.  The answer is "maybe".  If we passed by value we would have been forced to clone rather than reference secrets.  That might have helped.